_posts/2021-08-26-faas-storage-cassandra-astra.md

```diff
@@ -434,15 +433,6 @@ There's three ways to use the function:
 
 In production, you'll also want to add authentication to the submission endpoint. You can learn how with my Serverless For Everyone Else eBook listed at the end of the article. Here, I just want to focus on getting you connected and getting/putting documents into Astra DB.
 
-Check the table is empty:
```

Comment:
Why was this removed?

Reply:
The reason is that, in the Document API by Stargate (broadly modeled after other existing document databases), if the first action on a not-yet-existing collection is a write, the collection is automatically created for you; conversely, if the first action is a read, it fails with a 404 error.
This is a by-design behaviour of the underlying Stargate gateway which the `@astrajs/collections` SDK inherits: so, if the collection does not exist yet, this first read would fail with a 404.

I have to add that I did not know that beforehand, and had to inquiry with the engineering team to find out the rationale behind this choice; also, it is easy to overlook the not-found error on first pass (as happened to me on the first run) since it does not hinder anything of what follows (indeed the collection is created in the next step and everything proceeds flawlessy since then).

Reply:
Thanks for the explanation.

That is something that we should fix in the function then.. do you want to try revising the Node.js code?

The API (function) shouldn't leak the implementation of the document database underneath.

Reply:
In the Document API there is a specific endpoint for creating a collection, which goes something like this:
```
      curl -X POST \
        "https://<DB_ID>-<DB_REGION>.apps.astra.datastax.com/api/rest/v2/namespaces/functions/collections" \
        -H  "accept: application/json" \
        -H  "X-Cassandra-Token: <APP_TOKEN>" \
        -H  "Content-Type: application/json" \
        -d "{\"name\":\"COLLECTION_NAME\"}"
```
I am not sure this is exposed through the @astrajs/collections SDK, though (I can inquiry on this).

Adding the command above (to create "links" collection) to the setup instructions would make the functions work in all cases (i.e. even on a fresh install with no documents yet in the collection).

As said above, I'll look into the Node SDK about collection creation (it does not seem to be there, anyway).